Content.Server/Chat/LastMessageBeforeDeath.cs

```diff
@@ -0,0 +1,199 @@
+#if DEBUG
+            _sawmill = _logManager.GetSawmill("lastDeathMsgWebhook");
+#endif
+        }
+
+        // Public property to get the singleton instance
+        public static LastMessageBeforeDeath Instance
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    if (_instance == null)
+                    {
+                        _instance = new LastMessageBeforeDeath();
+                    }
+                    return _instance;
+                }
+            }
+        }
+
+        // Method to add a message for a player
+        public async void AddMessage(EntityUid source, string player, string message, string playerName)
+        {
+            if (message.Length > MaxICLength)
+            {
+                // if the message is bigger than the message length limit, we make it cut off at a random iterval.
+                // Example: Someone was giving a speech and got blown to bits.
+                int randomLength = _random.Next(1, MaxICLength);
+                message = message[..randomLength] + "-";
+            }
+
+            var currentTime = _gameTicker.RoundDuration();
+            string truncatedTime = $"{currentTime.Hours:D2}:{currentTime.Minutes:D2}:{currentTime.Seconds:D2}";
+            string formattedMessage = $"[{truncatedTime}] {playerName}: {message}";
+
+            if (!_playerData.Contains(player))
+            {
+                _playerData[player] = new OrderedDictionary();
+            }
+
+            var playerMessages = _playerData[player] as OrderedDictionary ?? throw new InvalidOperationException("Player messages dictionary is not initialized.");
+            playerMessages[playerName] = formattedMessage;
+            playerMessages["EntityUid"] = source;
+        }
+
+        // Send messages through the webhook on round end
+        public async void OnRoundEnd(WebhookIdentifier? webhookId)
+        {
+            if (!webhookId.HasValue)
+                return;
+
+            var id = webhookId.Value;
+            StringBuilder allMessagesString = new StringBuilder();
+            foreach (DictionaryEntry player in _playerData)
+            {
+                var playerMessages = player.Value as OrderedDictionary;
+                if (playerMessages != null)
+                {
+                    if (playerMessages["EntityUid"] is EntityUid playerUid && _mobStateSystem.IsDead(playerUid))
+                    {
+                        foreach (DictionaryEntry playerName in playerMessages)
+                        {
+                            if (playerName.Key.ToString() != "EntityUid")
+                            {
+                                allMessagesString.AppendLine($"{playerName.Value}");
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (allMessagesString.ToString().Length == 0)
```

Comment:
You can just iterate players here and get last messages then check bans without having to even add another event arg.

Reply:
Will do.